addons/knowledge/static/src/js/knowledge_controller.js

```diff
@@ -106,6 +107,29 @@ const KnowledgeArticleFormController = FormController.extend({
         await this._rename(id, event.currentTarget.value);
     },
 
+    /**
+     * When the user clicks on the name of the article, checks if the article
+     * name hasn't been set yet. If it hasn't, it will look for a title in the
+     * body of the article and set it as the name of the article.
+     * @param {Event} event
+     */
+    _onSetArticleName: async function (event) {
+        const name = event.currentTarget.value;
+        if (name === _t('New Article')) {
+            const $h1 = this.$('.o_knowledge_editor').find('h1')[0];
```

Comment:
Strange mix of jQuery and JS native code here.
You should use one or the other, e.g:

```js
// also please don't name variables "h1", the tag does not help understand the code
const $articleTitle = this.$('.o_knowledge_editor h1');
if ($articleTitle.length !== 0) {
    // would not use "event.currentTarget" here but instead use a selector to re-fetch the element you want
    // a bit more robust and understandable 
    // then also use jQuery for that (using the "val" method) for consistency
    event.currentTarget.value = $articleTitle.text();
    // ...
}